### PR TITLE
Replace substr with slice in ts files

### PR DESCRIPTION
**Pull request by nikoulai**
## Description

Replaces substr() with slice() in ts files.
No changed in **/lib/*.js files (should ?)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] I have selected the correct base branch.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran `npm run dtslint` with success and extended the tests and types if necessary.
- [ ] I ran `npm run test:cov` and my test cases cover all the lines and branches of the added code.
- [ ] I ran `npm run build` with success.
- [ ] I have tested the built `dist/web3.min.js` in a browser.
- [ ] I have tested my code on the live network.
- [ ] I have checked the Deploy Preview and it looks correct.
- [ ] I have updated the `CHANGELOG.md` file in the root folder.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -66,6 +66,7 @@
 		"cypress": "^9.6.0",
 		"cypress-jest-adapter": "^0.1.1",
 		"eslint": "^8.3.0",
+		"eslint-plugin-deprecation": "^1.3.2",
```
Comment (nazarhussain) [inline package.json]: Can we add this package to the `tools/eslint-config-web3-base/package.json`? May be other eslint plugins as well which are used in from that package.

**Review by nazarhussain — CHANGES_REQUESTED** [review]

**Review by avkos — APPROVED** [review]

**Review by jdevcs — APPROVED** [review]

File: packages/web3-validator/src/validation/bloom.ts [hunk 2]
```diff
@@ -64,7 +64,9 @@ export const isInBloom = (bloom: string, value: string | Uint8Array): boolean =>
 		// calculate bit position in bloom filter that must be active
 		const bitpos =
 			// eslint-disable-next-line no-bitwise
-			((parseInt(hash.substr(i, 2), 16) << 8) + parseInt(hash.substr(i + 2, 2), 16)) & 2047;
+			((parseInt(hash.slice(i, i + 2), 16) << 8) +
+				parseInt(hash.slice(i + 2, i + 2 + 2), 16)) &
```
Comment (jdevcs) [inline packages/web3-validator/src/validation/bloom.ts]: can be: `parseInt(hash.slice(i + 2, i + 4), 16)) &`

Comment (nikoulai) [reply packages/web3-validator/src/validation/bloom.ts]: I just did this way for clarity, if anyone read this. I'll change it and add a comment better.

**Review by Spacesai1or — APPROVED** [review]
LGTM besides Nazar and Junaid's feedback

File: packages/web3-utils/test/unit/random.test.ts [hunk 3]
```diff
@@ -24,6 +24,7 @@ describe('random library tests', () => {
 		describe('valid cases', () => {
 			it.each(randomHexData)('%s', input => {
 				const hexResult = randomHex(input);
+				// eslint-disable-next-line deprecation/deprecation
```
Comment (Spacesai1or) [inline packages/web3-utils/test/unit/random.test.ts:27]: What was the linter erroring about here?

Comment (nikoulai) [reply packages/web3-utils/test/unit/random.test.ts:27]: `isHexStrict` is marked [here](https://github.com/ChainSafe/web3.js/blob/f4cae280db3298b9e8e501a995083031c507dcf9/packages/web3-utils/src/validation.ts#L34) as deprecated and linter was throwing an error.

File: package.json (hunk 1 above)
Comment (nikoulai) [reply package.json]: Added ( + `eslint-plugin-header`) and removed from workspace root.

**Review by nazarhussain — APPROVED** [review]

**Review by luu-alex — APPROVED** [review]

**Pull request merged**